### PR TITLE
Added service to DAO module

### PR DESCRIPTION
**Pull request by Alexey007gold**

### CODE REVIEW

File: basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/entities/EntityInterface.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.alexkoveckiy.common.dao.entities;
+
+import java.io.Serializable;
+
+/**
+ * Created by alex on 11.03.17.
+ */
+public interface EntityInterface extends Serializable {
```
Comment (Dominys) [inline basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/entities/EntityInterface.java]: @Alexey007gold Why you need this empty interface ?

File: basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/repositories/TokenRepository.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.alexkoveckiy.common.dao.repositories;
+
+import com.alexkoveckiy.common.dao.entities.TokenEntity;
+import org.springframework.data.repository.PagingAndSortingRepository;
+
+/**
+ * Created by alex on 12.03.17.
+ */
+public interface TokenRepository extends PagingAndSortingRepository<TokenEntity, String> {
```
Comment (Dominys) [inline basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/repositories/TokenRepository.java]: @Alexey007gold All our entities will use String ids. So make some base repository.

File: basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/entities/TokenEntity.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import javax.persistence.Id;
+import javax.persistence.Table;
+import java.io.Serializable;
+
+/**
+ * Created by alex on 12.03.17.
+ */
+@Entity
+@Table(name = "tokens")
+public class TokenEntity implements EntityInterface {
```
Comment (Dominys) [inline basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/entities/TokenEntity.java]: @Alexey007gold We will not store tokens in DB.

File: basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/repositories/TokenRepository.java (hunk 2 above)
Comment (Alexey007gold) [reply basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/repositories/TokenRepository.java]: And spring will try to create realization for base repository!

File: basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/entities/EntityInterface.java (hunk 1 above)
Comment (Alexey007gold) [reply basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/entities/EntityInterface.java]: For generic BaseService

File: basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/repositories/TokenRepository.java (hunk 2 above)
Comment (Dominys) [reply basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/repositories/TokenRepository.java]: @Alexey007gold There is a way to solve this. Read documentation.

File: basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/entities/EntityInterface.java (hunk 1 above)
Comment (Dominys) [reply basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/entities/EntityInterface.java]: @Alexey007gold We don't need this.

File: basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/service/BaseService.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import com.alexkoveckiy.common.dao.entities.EntityInterface;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.repository.PagingAndSortingRepository;
+
+import java.io.Serializable;
+
+/**
+ * Created by alex on 11.03.17.
+ */
+public abstract class BaseService<E extends EntityInterface, R extends PagingAndSortingRepository<E, String>> {
```
Comment (Dominys) [inline basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/service/BaseService.java]: @Alexey007gold why pagingAndSorting and not your BaseRepository ?

File: basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/entities/DeviceEntity.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+ * Created by alex on 16.03.17.
+ */
+@Entity
+@Table(name = "devices")
+public class DeviceEntity implements Serializable {
+
+    private static final long serialVersionUID = 1115505457482755907L;
+
+    @Id
+    private String id;
```
Comment (Dominys) [inline basis/common-dao/src/main/java/com/alexkoveckiy/common/dao/entities/DeviceEntity.java]: @Alexey007gold All our entities will use String ID. make some base entity.

File: basis/common-dao/src/main/resources/changelog/changelog.xml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet author="Alex" id="changelog-1.0">
+        <createTable tableName="users">
+            <column name="id" type="VARCHAR(36)">
+                <constraints nullable="false" primaryKey="true" unique="true"/>
+            </column>
+            <column name="phoneNumber" type="TEXT">
```
Comment (Dominys) [inline basis/common-dao/src/main/resources/changelog/changelog.xml:14]: @Alexey007gold common naming pattern for fields in databases is lover-case with underscores.

File: basis/common-dao/src/main/resources/changelog/changelog.xml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+                <constraints nullable="false" primaryKey="false" unique="false"/>
+            </column>
+        </createTable>
+        <createTable tableName="devices">
+            <column name="id" type="VARCHAR(36)">
+                <constraints nullable="false" primaryKey="true" unique="true"/>
+            </column>
+            <column name="userId" type="TEXT">
+                <constraints nullable="false" primaryKey="false" unique="false"/>
```
Comment (Dominys) [inline basis/common-dao/src/main/resources/changelog/changelog.xml]: @Alexey007gold Why you need primaryKey="false" unique="false" ? It's default values.

**Comment by Dominys** [comment]
@Alexey007gold did I approve this ?

**Pull request merged**